### PR TITLE
Set particle host attribute slot and particle name in runtime.

### PR DESCRIPTION
**Pull request by shaper**
Does not yet cover slot sets or inner slots.

Despite this, it appears to be strictly an improvement over current slot/particle name instrumentation for illumination purposes, see sample screenshots ([previously](https://screenshot.googleplex.com/huM8fT0i6cW), [this patch](https://screenshot.googleplex.com/CGcnOKkcpUm)) and note the `action::Chooser` slot/particle is now instrumented.

This patch is an alternative to https://github.com/PolymerLabs/arcs/pull/1026 and is part of https://github.com/PolymerLabs/arcs/issues/1025.

### CODE REVIEW

File: runtime/slot.js [hunk 1]
```diff
@@ -48,8 +48,14 @@ class Slot {
   }
   startRender() {
     if (this.startRenderCallback) {
-      let contentTypes = this.constructRenderRequest();
-      this.startRenderCallback({particle: this.consumeConn.particle, slotName: this.consumeConn.name, contentTypes});
+      const slotName = this.consumeConn.name;
+      const particle = this.consumeConn.particle;
+      const context = this.getContext();
+      if (context.updateParticleName) {
```
Comment (shaper) [inline runtime/slot.js]: Maybe this is too much of a hack? This is to deal gracefully with whatever cases can result in our having a context here that's not a straight `DomContext`. Currently I know there's a `SetDomContext` and also some tests seem to pass `manifest`, and empty objects. If we'd prefer to do this differently am of course open to revision.

This PR is envisioned as a first step to restore illumination and move the slot/particle instrumentation needed into runtime, and I figure we can add sets, inner slots, etc. subsequently.
